server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileResource.java

```diff
@@ -63,16 +63,22 @@
   public static final String RESOURCE_SUFFIX = ".resource";
   static final String TEMP_SUFFIX = ".temp";
   private static final String CLOSING_SUFFIX = ".closing";
+  private static final int INIT_ARRAY_SIZE = 64;
```

Comment:
I'm not sure if this init array size is good or not. Please take a look.

Reply:
In the worst case, it may waste nearly half memory of `startTimes` and `endTimes`. From this point of view, I think it may be not a better choice than `Map<String, long[]> deviceToStartEndTime`.

Reply:
Here is an advice by @qiaojialin. When closing a resource, remove the redundant length of `startTimes` and `endTimes`. And when enlarging the arrays, use ` * 1.5` instead. 
By the way, the maximum number of devices in this TsFileResource shouldn't exceed the number of devices in the storage group. I'm looking for a way to find out this number.